File: binding/include/libqactiongroup.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qactiongroup.html
pub const qactiongroup = struct {

    /// New constructs a new QActionGroup object.
    pub fn New(parent: ?*anyopaque) QtC.QActionGroup {
        return qtc.QActionGroup_new(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QActionGroup_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qactiongroup.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn AddAction(self: ?*anyopaque, a: ?*anyopaque) QtC.QAction {
        return qtc.QActionGroup_AddAction(@ptrCast(self), @ptrCast(a));
    }

    pub fn AddAction2(self: ?*anyopaque, text: []const u8) QtC.QAction {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QActionGroup_AddAction2(@ptrCast(self), text_str);
    }

    pub fn AddAction3(self: ?*anyopaque, icon: ?*anyopaque, text: []const u8) QtC.QAction {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QActionGroup_AddAction3(@ptrCast(self), @ptrCast(icon), text_str);
    }

    pub fn RemoveAction(self: ?*anyopaque, a: ?*anyopaque) void {
        qtc.QActionGroup_RemoveAction(@ptrCast(self), @ptrCast(a));
    }

    pub fn Actions(self: ?*anyopaque, ) []const u8 {
        return qtc.QActionGroup_Actions(@ptrCast(self));
    }

    pub fn CheckedAction(self: ?*anyopaque, ) QtC.QAction {
        return qtc.QActionGroup_CheckedAction(@ptrCast(self));
    }

    pub fn IsExclusive(self: ?*anyopaque, ) bool {
        return qtc.QActionGroup_IsExclusive(@ptrCast(self));
    }

    pub fn IsEnabled(self: ?*anyopaque, ) bool {
        return qtc.QActionGroup_IsEnabled(@ptrCast(self));
    }

    pub fn IsVisible(self: ?*anyopaque, ) bool {
        return qtc.QActionGroup_IsVisible(@ptrCast(self));
    }

    pub fn ExclusionPolicy(self: ?*anyopaque, ) i32 {
        return qtc.QActionGroup_ExclusionPolicy(@ptrCast(self));
    }

    pub fn SetEnabled(self: ?*anyopaque, enabled: bool) void {
        qtc.QActionGroup_SetEnabled(@ptrCast(self), enabled);
    }

    pub fn SetDisabled(self: ?*anyopaque, b: bool) void {
        qtc.QActionGroup_SetDisabled(@ptrCast(self), b);
    }

    pub fn SetVisible(self: ?*anyopaque, visible: bool) void {
        qtc.QActionGroup_SetVisible(@ptrCast(self), visible);
    }

    pub fn SetExclusive(self: ?*anyopaque, exclusive: bool) void {
        qtc.QActionGroup_SetExclusive(@ptrCast(self), exclusive);
    }

    pub fn SetExclusionPolicy(self: ?*anyopaque, policy: i32) void {
        qtc.QActionGroup_SetExclusionPolicy(@ptrCast(self), @intCast(policy));
    }

    pub fn Triggered(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QActionGroup_Triggered(@ptrCast(self), @ptrCast(param1));
    }

    pub fn OnTriggered(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QActionGroup_Connect_Triggered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Hovered(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QActionGroup_Hovered(@ptrCast(self), @ptrCast(param1));
    }

    pub fn OnHovered(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QActionGroup_Connect_Hovered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QActionGroup_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qactiongroup.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QActionGroup_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qactiongroup.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QActionGroup_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/actiongroup.html#types
pub const enums = struct {
};

File: binding/src/libqactiongroup.cpp
#include <QAction>
#include <QActionGroup>
#include <QIcon>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qactiongroup.h>
#include "libqactiongroup.h"
#include "libqactiongroup.hxx"

QActionGroup* QActionGroup_new(QObject* parent) {
	 return new QActionGroup(parent);
}

libqt_string QActionGroup_Tr(const char* s) {
	QString _ret = QActionGroup::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAction* QActionGroup_AddAction(QActionGroup* self, QAction* a) {
	return self->addAction(a);
}

QAction* QActionGroup_AddAction2(QActionGroup* self, const libqt_string text) {
	return self->addAction(QString::fromUtf8(text.data, text.len));
}

QAction* QActionGroup_AddAction3(QActionGroup* self, const QIcon* icon, const libqt_string text) {
	return self->addAction(*icon, QString::fromUtf8(text.data, text.len));
}

void QActionGroup_RemoveAction(QActionGroup* self, QAction* a) {
	self->removeAction(a);
}

libqt_list QActionGroup_Actions(const QActionGroup* self) {
	auto _ret = self->actions();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

QAction* QActionGroup_CheckedAction(const QActionGroup* self) {
	return self->checkedAction();
}

bool QActionGroup_IsExclusive(const QActionGroup* self) {
	return self->isExclusive();
}

bool QActionGroup_IsEnabled(const QActionGroup* self) {
	return self->isEnabled();
}

bool QActionGroup_IsVisible(const QActionGroup* self) {
	return self->isVisible();
}

int QActionGroup_ExclusionPolicy(const QActionGroup* self) {
	return self->exclusionPolicy();
}

void QActionGroup_SetEnabled(QActionGroup* self, bool enabled) {
	self->setEnabled(enabled);
}

void QActionGroup_SetDisabled(QActionGroup* self, bool b) {
	self->setDisabled(b);
}

void QActionGroup_SetVisible(QActionGroup* self, bool visible) {
	self->setVisible(visible);
}

void QActionGroup_SetExclusive(QActionGroup* self, bool exclusive) {
	self->setExclusive(exclusive);
}

void QActionGroup_SetExclusionPolicy(QActionGroup* self, int policy) {
	self->setExclusionPolicy(static_cast<QActionGroup::ExclusionPolicy>(policy));
}

void QActionGroup_Triggered(QActionGroup* self, QAction* param1) {
	self->triggered(param1);
}

void QActionGroup_Connect_Triggered(QActionGroup* self, intptr_t slot) {
    void (*slotFunc)(QActionGroup*, QAction*) = reinterpret_cast<void (*)(QActionGroup*, QAction*)>(slot);
    QActionGroup::connect(self, &QActionGroup::triggered, [self, slotFunc](QAction* param1) {
	slotFunc(self, param1);
    });
}

void QActionGroup_Hovered(QActionGroup* self, QAction* param1) {
	self->hovered(param1);
}

void QActionGroup_Connect_Hovered(QActionGroup* self, intptr_t slot) {
    void (*slotFunc)(QActionGroup*, QAction*) = reinterpret_cast<void (*)(QActionGroup*, QAction*)>(slot);
    QActionGroup::connect(self, &QActionGroup::hovered, [self, slotFunc](QAction* param1) {
	slotFunc(self, param1);
    });
}

libqt_string QActionGroup_Tr2(const char* s, const char* c) {
	QString _ret = QActionGroup::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QActionGroup_Tr3(const char* s, const char* c, int n) {
	QString _ret = QActionGroup::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QActionGroup_Delete(QActionGroup* self) {
    delete self;
}

